File: frameworks/dotnet/wolverine-http/UnitTests/MiddlewareTests.cs
namespace UnitTests;

[Collection(nameof(WolverineApp))]
public sealed class MiddlewareTests(WolverineApp app)
{
    // rb:test middleware.none,middleware.four,middleware.sixteen
    [Theory]
    [Trait("corpus", "middleware.none")]
    [Trait("corpus", "middleware.four")]
    [Trait("corpus", "middleware.sixteen")]
    [InlineData("none")]
    [InlineData("four")]
    [InlineData("sixteen")]
    public async Task Every_layer_passes_the_request_through(string layers)
    {
        IScenarioResult result = await app.Host.Scenario(s => s.Get.Url($"/middleware/{layers}"));

        Answer.Is(Expected.Json("items.small.json"), result);
    }
}

File: frameworks/java/vertx/UnitTests/src/test/java/unittests/MiddlewareTests.java
package unittests;

import org.junit.jupiter.api.Tag;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class MiddlewareTests extends VertxApp {

    // rb:test middleware.none,middleware.four,middleware.sixteen
    @ParameterizedTest
    @Tag("middleware.none")
    @Tag("middleware.four")
    @Tag("middleware.sixteen")
    @ValueSource(strings = {"none", "four", "sixteen"})
    void everyLayerPassesTheRequestThrough(String layers) throws Exception {
        Answer.is(Expected.json("items.small.json"), get("/middleware/" + layers));
    }
}
